File: app/src/main/java/com/digicoffer/lauditor/AuditTrails/Adapters/PaginationHelper.java
package com.digicoffer.lauditor.AuditTrails.Adapters;

public class PaginationHelper {
    public static int calculateTotalNoOfPages(int itemcount, int itemsperpage) {

        return (int) Math.ceil(((double) itemcount) / (double) (itemsperpage));
    }

    public static int startIndexForCurrentPage(int CurrentPageNo, int itemsperpage) {
        int StartIndex = (CurrentPageNo - 1) * itemsperpage;
        return StartIndex;
    }

    public static int endIndexForCurrentPage(int StartIndex, int ItemCount, int itemsperpage) {
        int EndIndex = Math.min(StartIndex + itemsperpage, ItemCount);
        return EndIndex;
    }

}

File: app/src/main/java/com/digicoffer/lauditor/AuditTrails/Adapters/PaginationHelper.kt
package com.digicoffer.lauditor.AuditTrails.Adapters

import kotlin.math.ceil

object PaginationHelper {
    @JvmStatic
    fun calculateTotalNoOfPages(itemcount: Int, itemsperpage: Int): Int {
        return ceil(itemcount.toDouble() / itemsperpage.toDouble()).toInt()
    }

    @JvmStatic
    fun startIndexForCurrentPage(CurrentPageNo: Int, itemsperpage: Int): Int {
        return (CurrentPageNo - 1) * itemsperpage
    }

    @JvmStatic
    fun endIndexForCurrentPage(StartIndex: Int, ItemCount: Int, itemsperpage: Int): Int {
        return Math.min(StartIndex + itemsperpage, ItemCount)
    }
}
